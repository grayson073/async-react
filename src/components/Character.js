import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  const { name, status, species, gender } = character;
  return (
    <tr>
      <td>{name}</td>
      <td>{status}</td>
      <td>{species}</td>
      <td>{gender}</td>
    </tr>
  );
}

Character.propTypes = {
  character: PropTypes.object.isRequired
};

export default Character;
