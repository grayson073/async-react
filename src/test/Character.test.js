import React from 'react';
import renderer from 'react-test-renderer';
import Character from '../components/Character';

describe('Character', () => {
  
  it('Renders correctly', () => {
    const character = {
      name: 'Frank',
      status: 'alive',
      species: 'dog',
      gender: 'male'
    };

    const tree = renderer.create(
      <Character character={character}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
