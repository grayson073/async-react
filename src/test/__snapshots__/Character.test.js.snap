// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Renders correctly 1`] = `
<tr>
  <td>
    Frank
  </td>
  <td>
    alive
  </td>
  <td>
    dog
  </td>
  <td>
    male
  </td>
</tr>
`;
